core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -404,19 +386,34 @@ public List<ActiveCompaction> getActiveCompactions()
   @Override
   public List<ActiveCompaction> getActiveCompactions(Collection<ServerId> compactionServers)
       throws AccumuloException, AccumuloSecurityException {
+    return queryServers(compactionServers, this::getActiveCompactions,
+        INSTANCE_OPS_COMPACTIONS_FINDER_POOL);
+  }
+
+  private <T> List<T> queryServers(Collection<ServerId> servers, ServerQuery<List<T>> serverQuery,
+      ThreadPoolNames pool) throws AccumuloException, AccumuloSecurityException {
+
+    final ExecutorService executorService;
+    // If size 0 or 1 there's no need to create a thread pool
+    if (servers.isEmpty()) {
+      return List.of();
+    } else if (servers.size() == 1) {
+      executorService = MoreExecutors.newDirectExecutorService();
+    } else {
+      int numThreads = Math.max(4, Math.min((servers.size()) / 10, 256));
```

Comment:
Am I reading that correctly, that this will create 1 thread for every 10 servers where the server count is between 41 and 2560? I think 256 threads in the client might be too much.

Reply:
It could be too many but I am not sure, I refactored this code to be shared for getting scans and compations. getCompactions() previously was already creating up to 256 threads so it came from that: https://github.com/apache/accumulo/blob/c6ec9697da2cfa517eb028d8937ecf10b31c9d6a/core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java#L408

We could lower the number to something smaller for both.

Reply:
ok, I didn't realize that we were already doing this.

Reply:
I added a follow on https://github.com/apache/accumulo/issues/5135 to look into setting a lower default or adding a property

Reply:
256 threads is probably ok when there is nothing else going on in the process.  It could cause problems for a process that already has a lot of threads though.